packages/flutter/test/widgets/list_view_viewporting_test.dart

```diff
@@ -429,4 +429,36 @@ void main() {
     expect(position.viewportDimension, equals(600.0));
     expect(position.minScrollExtent, equals(0.0));
   });
+
+  testWidgets('ListView should not paint hidden children', (WidgetTester tester) async {
+    await tester.pumpWidget(
+        new Directionality(
+            textDirection: TextDirection.ltr,
+            child: new Center(
+              child: new Container(
+                  height: 200.0,
+                  child: new ListView(
+                    cacheExtent: 500.0,
+                    controller: new ScrollController(initialScrollOffset: 300.0),
+                    children: <Widget>[
+                      new Container(height: 140.0),
+                      new Container(height: 160.0),
+                      new Container(height: 90.0),
+                      new Container(height: 110.0),
+                      new Container(height: 80.0),
+                      new Container(height: 70.0),
+                    ],
+                  )
+              ),
+            )
+        )
+    );
+
+    final RenderSliverList list = tester.renderObject(find.byType(SliverList));
+    expect(list, isNotNull);
+    expect(list, hasAGoodToStringDeep);
+    final String dump = list.toStringDeep(minLevel: DiagnosticLevel.info);
+    final int numChildrenPainted = 'layer: OffsetLayer'.allMatches(dump).length;
+    expect(numChildrenPainted, 2);  // only the middle two should be painted
```

Comment:
this test might be a good candidate for the mock_canvas API (search for tests with expectations that mention `paints`)

Reply:
@goderbauer The SliverGrid can only scroll along the main axis, and this PR would prune those that are outside of the main axis. But this PR won't prune widgets that are outside of the cross axis. I guess that there won't be many hidden widgets along the cross axis because you can never scroll to reveal them. Hence we probably don't need to prune them.